src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelist.java

```diff
@@ -147,6 +148,13 @@ public final void reset(Collection<? extends Whitelist> delegates) {
         }
     }
 
+    private void addWrapper(ProxyWhitelist pwl){
+        lock.writeLock().lock();
+        wrappers.put(pwl, null);
+        lock.writeLock().unlock();
```

Comment:
Even if we don't expect `WeakHashMap#put` to throw an exception, I think it would be a good idea to use a `try`/`finally` block for the lock.

Reply:
I agree -- this is a clear 🐛  for me, because an unexpected exception can prevent us ever unlocking.

Reply:
Agree, I will add it.

Reply:
[sigh](https://stackoverflow.com/a/17043350/12916)